### ISSUE TITLE
[BUG] Inngest break build with NextJs 15 canary 177 

### ISSUE THREAD

**Issue by thibaultleouay** [inngest/inngest-js#717]
**Describe the bug**
I'm trying to build a next 15 canary project with Inngest.  

**To Reproduce**
Example repo 

https://github.com/thibaultleouay/inngest-nextjs15-bug

and run pnpm build

**Expected behavior**
A successful build

**Code snippets / Logs / Screenshots**
```
src/app/api/inngest/route.ts
Type error: Route "src/app/api/inngest/route.ts" has an invalid "GET" export:
  Type "ServerResponse<IncomingMessage> & { send: Send<any>; json: Send<any>; ... 5 more ...; revalidate: (urlPath: string, opts?: { ...; } | undefined) => Promise<...>; }" is not a valid type for the function's second argument.
```

**Comment by thibaultleouay** [comment]
This is ugly, but I can solve it like this.
```
import { serve } from "inngest/next";
import { inngest } from "../../../inngest/client";
import { helloWorld } from "../../../inngest/functions";
import type { NextRequest } from "next/server";

// Create an API that serves zero functions
 const s = serve({
  client: inngest,
  functions: [
    helloWorld,
    /* your functions will be passed here later! */
  ],
});

const GET = s.GET as (expectedReq: NextRequest) => Promise<Response>;
const POST = s.POST as (expectedReq: NextRequest) => Promise<Response>
const PUT = s.PUT as (expectedReq: NextRequest) => Promise<Response>

export { GET,  POST, PUT };

```

**Comment by jpwilliams** [comment]
Thanks for the report, @thibaultleouay!

There's a fix up over at #721 - let me know if this resolves your issue! :)

**Comment by thibaultleouay** [comment]
@jpwilliams  just tried it wiht #721  and it works 🙏
